Core/include/Acts/TrackFinding/CombinatorialKalmanFilter.hpp

```diff
@@ -73,14 +73,21 @@ struct CombinatorialKalmanFilterExtensions {
   using candidate_container_t =
       typename std::vector<typename traj_t::TrackStateProxy>;
 
+  enum class BranchStopperResult {
```

Comment:
It's a bit awkward that this `enum` that is needed by user code is inside a templated class (`Acts::CombinatorialKalmanFilterExtensions<traj_t>::BranchStopperResult`), even though the template parameter isn't relevant for the definition.

Would it make sense to move it out as `Acts::CombinatorialKalmanFilterBranchStopperResult`?